### PR TITLE
util: added umask manipulation

### PR DESCRIPTION
**Pull request by matrixhead**
This pr tries to add umask manipulation to test util, this would allow changing umask in a contained manner instead of doing it globally.

### CODE REVIEW

**Comment by matrixhead** [comment]
@tertsdiepraam Does this look like it will do the trick? Should I port existing tests to utilize this?

**Comment by tertsdiepraam** [comment]
I love it!

File: tests/common/util.rs [hunk 1]
```diff
@@ -3872,4 +3894,21 @@ mod tests {
             .no_stderr()
             .stdout_is("8\n16\n");
     }
+
+    #[cfg(unix)]
+    #[test]
+    fn test_altering_umask() {
+        use uucore::mode::get_umask;
+        let p_umask = get_umask();
+        //make sure we are not testing against the same umask
```
Comment (tertsdiepraam) [inline tests/common/util.rs:3903]:
```suggestion
        // make sure we are not testing against the same umask
```
Just a tiny suggestion :)

**Comment by tertsdiepraam** [comment]
And yes, it'd be great if you could port a few tests (in a separate commit) so that we can see that this actually solves our issue.

**Comment by sylvestre** [comment]
thanks!

**Pull request merged**